### PR TITLE
F #-: Lithops appliance

### PR DESCRIPTION
**Pull request by Aletibazo**
Created new appliance for Lithops in the appliances/Community folder, to be included on the Community Marketplace.

This appliance installs Lithops v3.4.0 on Ubuntu 22.04, with Docker v5:26.1.3-1. It has support for localhost compute backend and localhost / minio storage backend

### CODE REVIEW

**Review by tinova — APPROVED** [review]

File: appliances/Community/Lithops/appliance.sh [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+    msg info "Install Lithops from pip"
+    if ! pip install lithops==${LITHOPS_VERSION} ; then
+        msg error "Error installing Lithops"
+        exit 1
+    fi
+
+    msg info "Create /etc/lithops folder"
+    mkdir /etc/lithops
+
+    return $?
```
Comment (sk4zuzu) [inline appliances/Community/Lithops/appliance.sh:199]:
```suggestion
    return 0
```

or I think the whole return statement can be just removed :thinking:

File: appliances/Community/Lithops/appliance.sh [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+        else
+            msg info "Adding MinIO configuration to /etc/lithops/config"
+            sed -i -ne "/# Start Storage/ {p; iminio:\n  endpoint: ${ONEAPP_MINIO_ENDPOINT}\n  access_key_id: ${ONEAPP_MINIO_ACCESS_KEY_ID}\n  secret_access_key: ${ONEAPP_MINIO_SECRET_ACCESS_KEY}\n  storage_bucket: ${ONEAPP_MINIO_BUCKETT}" -e ":a; n; /# End Storage/ {p; b}; ba}; p" /etc/lithops/config
+        fi
+    fi
+}
+
+check_minio_attrs()
+{
+    [ -z "$ONEAPP_MINIO_ENDPOINT" ] && return 1
```
Comment (sk4zuzu) [inline appliances/Community/Lithops/appliance.sh:242]: For consistency:

```suggestion
    [[ -z "$ONEAPP_MINIO_ENDPOINT" ]] && return 1
```

also in other lines in this function.

File: appliances/Community/Lithops/appliance.sh [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+- ONEAPP_MINIO_ENDPOINT_CERT is necessary when using self-signed certificates on the MinIO server. This is the
+certificate for the CA on the MinIO server. If the CA certificate exists, script will skip it,
+if one would want to update the CA certificate from context, first delete previous ca.crt file.
+EOF
+)
+ONE_SERVICE_RECONFIGURABLE=true
+
+### Contextualization defaults #######################################
+
+ONEAPP_BACKEND="${ONEAPP_BACKEND:-localhost}"
```
Comment (sk4zuzu) [inline appliances/Community/Lithops/appliance.sh:59]: Maybe ONEAPP_BACKEND and ONEAPP_STORAGE could be called something like ONEAPP_LITHOPS_BACKEND and ONEAPP_LITHOPS_STORAGE, so later when somebody configures this in for example a flow service (with other apps), there are no name conflicts? :thinking:

File: appliances/Community/Lithops/appliance.sh [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+
+    # cleanup
+    postinstall_cleanup
+
+    msg info "INSTALLATION FINISHED"
+
+    return 0
+}
+
+service_configure()
```
Comment (sk4zuzu) [inline appliances/Community/Lithops/appliance.sh:112]: I'm not sure what exactly gets executed, so docker is installed, some pip packages, but how and where all this is used? :)

**Review by sk4zuzu — CHANGES_REQUESTED** [review]

**Review by sk4zuzu — APPROVED** [review]

**Pull request closed without merging**